modules/KoreBuild.Tasks/CodeSign.targets

```diff
@@ -0,0 +1,43 @@
+<Project>
+  <PropertyGroup>
+    <RoslynSignToolPath>$(MSBuildThisFileDirectory)SignTool\SignTool.exe</RoslynSignToolPath>
+    <SkipCodeSign Condition=" '$(OS)' != 'Windows_NT' ">true</SkipCodeSign>
+
+    <_IntermediateSignToolDataFile>$(IntermediateDir)SignToolData.g.json</_IntermediateSignToolDataFile>
+
+    <!-- Repos can explicitly define the signdata.json file if they want do. They must set SignToolDataFile to the path of the file. When set, file generation is skipped. -->
+    <GenerateSignToolDataFile  Condition="'$(SignToolDataFile)' == '' ">true</GenerateSignToolDataFile>
+    <SignToolDataFile Condition="'$(GenerateSignToolDataFile)' == 'true'">$(_IntermediateSignToolDataFile)</SignToolDataFile>
+  </PropertyGroup>
+
+  <Target Name="CodeSign" Condition=" '$(SkipCodeSign)' != 'true' "
+          AfterTargets="Package"
+          DependsOnTargets="GetArtifactInfo">
+
+    <PropertyGroup>
+    </PropertyGroup>
+
+    <GetPathToFullMSBuild>
+      <Output TaskParameter="MSBuildx86Path" PropertyName="MSBuildx86Path" />
+    </GetPathToFullMSBuild>
+
+    <GenerateSignToolDataFile Condition=" '$(GenerateSignToolDataFile)' != 'false' "
+      Files="@(FilesToSign)"
+      Exclusions="@(FilesToExcludeFromSigning)"
+      OutputPath="$(_IntermediateSignToolDataFile)" />
+
+    <ItemGroup>
+      <SignToolOptions Condition=" '$(MSBuildx86Path)' != '' " Include="-msbuildPath;$(MSBuildx86Path)" />
+      <SignToolOptions Condition=" '$(SignType)' != 'real' AND '$(SignType)' != 'test' "  Include="-test" />
```

Comment:
Just curious about when we use SignType != test or real, some kind of other special test case?

Reply:
SignType!= test or real for local dev builds. This instructs signtool to do a dry-run without actual codesigning. The tool does some validation of the signing configuration and should catch the most common errors.